### PR TITLE
[dev] feat(moe): Support apply wd to qk layernorm for Qwen3-Next

### PR DESCRIPTION
**Pull request by yuzhongw-nvidia**
# What does this PR do ?

:warning: For major changes (either in lines of code or in its impact), please make sure to first share a design doc with the team. If you're unsure what's the best way to do so, contact the @megatron-oncall.

[PR for main](https://github.com/NVIDIA/Megatron-LM/pull/2753)

## Contribution process

```mermaid
flowchart LR
    A[Pre-checks] --> B[PR Tests]
    subgraph Code Review/Approval
        C1[Expert Review] --> C2[Final Review]
    end
    B --> C1
    C2 --> D[Merge]
```

### Pre-checks

- [ ] I want this PR in a versioned release and have added the appropriate Milestone (e.g., `Core 0.8`)
- [ ] I have added relevant unit tests
- [ ] I have added relevant functional tests
- [ ] I have added proper typing to my code [Typing guidelines](https://docs.python.org/3/library/typing.html)
- [ ] I have added relevant documentation
- [ ] I have run the [autoformatter.sh](https://github.com/NVIDIA/Megatron-LM/blob/main/tools/autoformat.sh) on my PR

### Code review

The following process is enforced via the CODEOWNERS file for changes into `megatron/core`. For changes outside of `megatron/core`, it is up to the PR author whether or not to tag the Final Reviewer team.

<details>
<summary>For MRs into `main` branch</summary>

Feel free to message or comment the @megatron-oncall to help accelerate your merge into main. The less complex your PR is, the faster it will be approved and merged!

#### (Step 2): Collect the expert reviewers reviews

1. Attach the `Expert Review` label when your PR is ready for review.
2. GitHub auto-assigns expert reviewers based on your changes. They will get notified and pick up your PR soon.

:warning: Only proceed to the next step once all reviewers have approved, merge-conflict are resolved and the CI is passing.  
Final Review might get declined if these requirements are not fulfilled.

#### (Step 3): Final Review

1. Add `Final Review` label
2. GitHub auto-assigns final reviewers based on your changes. They will get notified and pick up your PR soon.

#### (Optional Step 4): Cherry-pick into release branch

If this PR also needs to be merged into `core_r*` release branches, after this PR has been merged, select `Cherry-pick` to open a new PR into the release branch.

</details>

<details>
<summary>For MRs into `dev` branch</summary>
The proposed review process for `dev` branch is under active discussion.

MRs are mergable after one approval by either `eharper@nvidia.com` or `zijiey@nvidia.com`.
</details>

### Merging your PR

Any member of [core-adlr](https://github.com/orgs/teams/NVIDIA/core-adlr) and [`core-nemo`](https://github.com/orgs/teams/NVIDIA/core-nemo) will be able to merge your PR.

### CODE REVIEW

File: megatron/core/ssm/gated_delta_net.py [hunk 1]
```diff
@@ -246,7 +246,8 @@ def reset_parameters(self):
                     dtype=self.config.params_dtype,
                     device=torch.cuda.current_device(),
                 ).uniform_(*self.A_init_range)
-                self.A_log.data.copy_(A)
+                A_log = torch.log(A)
+                self.A_log.data.copy_(A_log)
```
Comment (BestJuly) [inline megatron/core/ssm/gated_delta_net.py]: `self.A_log.data.copy_(torch.log(A))` would be preferred if `A_log` will not be used somewhere else.

File: tests/unit_tests/test_optimizer.py [hunk 2]
```diff
@@ -587,3 +587,74 @@ def test_get_megatron_optimizer_custom_process_groups_validation():
...
+            wd_mult_1_count += num_params
+        else:
+            wd_mult_0_count += num_params
+
+    # Expected:
+    # wd_mult=1.0: q_layernorm.weight, q_layernorm.bias, k_layernorm.weight,
+    #              k_layernorm.bias, linear.weight = 5 params
+    # wd_mult=0.0: regular_layernorm.weight, linear.bias = 2 params
+    assert wd_mult_1_count == 5, f"Expected 5 params with wd_mult=1.0, but got {wd_mult_1_count}"
+    assert wd_mult_0_count == 2, f"Expected 3 params with wd_mult=0.0, but got {wd_mult_0_count}"
```
Comment (BestJuly) [inline tests/unit_tests/test_optimizer.py]: 3 -> 2

File: megatron/core/optimizer/optimizer_config.py [hunk 3]
```diff
@@ -40,6 +40,15 @@ class OptimizerConfig:
     weight_decay: float = 0.01
     """Weight decay coefficient for L2 regularization."""
 
+    no_weight_decay_cond: Optional[str] = None
```
Comment (BestJuly) [inline megatron/core/optimizer/optimizer_config.py]: The argument is `no_weight_decay_cond_type`. We should make it consistent.
